Task3/Queue/QueueExtensions.cs

```diff
@@ -0,0 +1,28 @@
+﻿namespace Queue
+{
+    public static class QueueExtensions
+    {
+        public static IQueue<T> Tail<T>(this IQueue<T> queue) where T : struct
+        {
+            var tempQueue = queue;
+            var newQueue = new Queue<T>();
+            bool isFirst = true;
+
+            if (queue.IsEmpty())
+            {
+                throw new InvalidOperationException("Queue is empty!");
+            }
+
+            while (!tempQueue.IsEmpty())
+            {
```

Comment:
I am not sure whether using reflection would be overkill(for this task) in order to deep clone the queue. Because of that I think that it would be also valid solution to make a tempQueue which will store all of the original's queue data and then to restore the data after all other operations are made